array-manipulation-3/src/equal.js

```diff
@@ -0,0 +1,17 @@
+/* exported equal */
+function equal(first, second) {
+  var isTrue = true;
+  if (first.length === second.length) {
+    for (var i = 0; i < first.length; i++) {
+      if (first[i] === second[i]) {
+        isTrue = true;
+      } else {
+        isTrue = false;
+        break;
+      }
+    }
+  } else {
+    isTrue = false;
+  }
+  return isTrue;
```

Comment:
You can simplify this quite a bit with a "guard" right at the top of the function that returns `false` if the lengths are not equal. If during any iteration of the loop the values in the array are not equal, return `false` immediately. If the loop completes (and therefore never returned), then return `true`.